relayer/src/link.rs

```diff
@@ -1426,34 +1429,42 @@ impl RelayPath {
         Ok(())
     }
 
-    /// Pulls out the operational elements with elapsed delay period and that can
+    /// Destination-chain fetching of operational data.
+    ///
+    /// Pulls out an operational elements with elapsed delay period and that can
     /// now be processed. Does not block: if no OD fulfilled the delay period (or none is
-    /// scheduled), returns immediately with `vec![]`.
-    fn try_fetch_scheduled_operational_data(
```

Comment:
To explain the changes here at a high level, since the diff is a big messy:

I split the method `try_fetch_scheduled_operational_data` into two separate methods:

1. `try_fetch_scheduled_operational_data_dst` which fetches 1 op. data specifically for the destination chain
2. `try_fetch_scheduled_operational_data_src` which fetches 1 op. data specifically for the source chain.